xbmc/cores/DataCacheCore.h

```diff
@@ -195,8 +198,23 @@ class CDataCacheCore
    */
   int64_t GetMaxTime();
 
+  /*!
+   * \brief Cache the current playing item
+   *
+   * \param item - the item to store
+   */
+  void SetFileItem(const CFileItem& item);
+
+  /*!
+   * \brief Get the current playing item
+   *
+   * \return the item currently cached by the player
+   */
+  const std::shared_ptr<CFileItem> GetFileItem() const;
```

Comment:
The `const` on the `std::shared_ptr` doesn't make sense here, it's returned by value so the `const` doesn't have an effect. What you probably want is `std::shared_ptr<const CFileItem> GetFileItem() const;` but that requires const-correctness fixes in a number of files.

Reply:
could be the intention is to return a const reference to the shared_ptr. in general we should not return plain const of any type

Reply:
Can't be a const ref as the item might be a nullptr if not set. I'll drop the const qualifier, it doesn't make sense on return.